src/components/Dropdown.tsx

```diff
@@ -2,37 +2,31 @@ import './styles/dropdown.less'
 
 import React from 'react'
 import ReactDOM from 'react-dom'
-import PropTypes from 'prop-types'
 import classNames from 'classnames'
 
-/**
- * @typedef {Object} BoxProps
- * @property {() => void} [onShowBody]
- * @property {() => void} [onHideBody]
- * @property {boolean} [allowBodyInteractions]
- * @property {boolean} [top]
- * @property {boolean} [right]
- * @property {string} [scrolling_parent]
- * @property {[React.ReactElement<TriggerProps>, React.ReactElement<{}> | ((props: {}) => JSX.Element)]} children
- * @property {string} [className]
- */
-
-/**
- * @typedef {Object} BoxState
- * @typedef {boolean} top
- * @typedef {boolean} show_body
- */
-
-/** @extends {React.Component<BoxProps, BoxState>} */
-class Box extends React.Component<any, any> {
-    public static displayName
-    public static propTypes
-
-    /**
-     * @param {BoxProps} props
-     * @param {unknown} context
-     */
-    constructor(props, context) {
+interface BoxProps {
+    onShowBody?: () => void
+    onHideBody?: () => void
+    allowBodyInteractions?: boolean
+    top?: boolean
+    right?: boolean
+    scrolling_parent?: string
+    children?: [
+        React.ReactElement<TriggerProps>,
+        React.ReactElement<{}> | ((props: {}) => JSX.Element)
+    ]
+    className?: string
+}
+
+interface BoxState {
+    top: boolean
+    show_body: boolean
+}
+
+class Box extends React.Component<BoxProps, BoxState> {
+    public static displayName: string
+
+    constructor(props: BoxProps, context: React.Context<any>) {
```

Comment:
@vishnugopal We've discussed it in the past and it seems that if `context` is unused we can omit it. If it is needed, perhaps it can be `Context<unknown>` (safer)?

Reply:
Makes sense! Done!